applications/StructuralMechanicsApplication/custom_elements/truss_element_linear_3D2N.cpp

```diff
@@ -187,6 +211,79 @@ void TrussElementLinear3D2N::WriteTransformationCoordinates(
   KRATOS_CATCH("");
 }
 
+double TrussElementLinear3D2N::CalculateLinearStrain()  {
+  KRATOS_TRY;
+
+  Vector current_disp = ZeroVector(msLocalSize);
+  this->GetValuesVector(current_disp);
+  BoundedMatrix<double, msLocalSize, msLocalSize> transformation_matrix =
+      ZeroMatrix(msLocalSize, msLocalSize);
+  this->CreateTransformationMatrix(transformation_matrix);
+
+  current_disp = prod(Matrix(trans(transformation_matrix)),current_disp);
+  const double length_0 = this->CalculateReferenceLength();
+  const double e = (current_disp[3]-current_disp[0])/length_0;
+
+  return e;
+  KRATOS_CATCH("");
+}
+
+
+void TrussElementLinear3D2N::UpdateInternalForces(BoundedVector<double,msLocalSize>& rinternalForces)
+{
+  KRATOS_TRY;
+
+  Vector temp_internal_stresses = ZeroVector(msLocalSize);
+  ProcessInfo temp_process_information;
+  ConstitutiveLaw::Parameters Values(this->GetGeometry(),this->GetProperties(),temp_process_information);
+  this->mConstitutiveLaw->CalculateValue(Values,NORMAL_STRESS,temp_internal_stresses);  
+
+  rinternalForces = temp_internal_stresses*this->GetProperties()[CROSS_AREA];
+
+
+  BoundedMatrix<double, msLocalSize, msLocalSize> transformation_matrix =
+      ZeroMatrix(msLocalSize, msLocalSize);
+  this->CreateTransformationMatrix(transformation_matrix);
+
+  rinternalForces = prod(transformation_matrix, rinternalForces);

```

Comment:
noalies?

Reply:
noalias is still strange to me. Adding it here makes my tests crush ????